### PR TITLE
Issue #4394: united api and filters profiles, increased coverage to 85%

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4394

Profiles pitest-checkstyle-api and pitest-checkstyle-filters were united in order to increase coverage of api profile, as they have many common functionality. More then with any other profile. 
Coverage of api profile before union: 67% test coverage and 45% mutation coverage.
Coverage of api profile after union: 92% test coverage and 75% mutation coverage.

[pitest-checkstyle-api profile before changes](https://nimfadora.github.io/api.html)
[pitest-checkstyle-api-filters profile after union before coverage increasing changes](https://nimfadora.github.io/filters_union_api.html)

increased mutation threshold for pitest-checkstyle-api-filters profile by 4%
current threshold: 85%
execution time: 9:51 min

### CODE REVIEW

**Comment by romani** [comment]
@Nimfadora , please fix all CIs.
You missed ":" in commit message.
Please provide pitest reports before changes.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbsttractFileSetCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.List;
+import java.util.SortedSet;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.Checker;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+
+public class AbsttractFileSetCheckTest {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbsttractFileSetCheckTest.java]: Too many `t`s.
`Absttract` => `Abstract`

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractCheckTest.java [hunk 2]
```diff
@@ -73,4 +78,65 @@ public void testVisitToken() {
...
     }
+
+    @Test
+    public void testGetLine() throws Exception {
+        final AbstractCheck check = new AbstractCheck() {
+            @Override
+            public int[] getDefaultTokens() {
+                return CommonUtils.EMPTY_INT_ARRAY;
+            }
+
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractCheckTest.java]: Please remove these empty lines before close `}`. There are a few in tests you have made.

**Comment by Nimfadora** [comment]
@rnveach @romani, after rebasing on the latest master test JavadocParseTreeTest#testInlineCustomJavadocTag()
test failed. Git says that this change was introduced by Sharma

**Comment by romani** [comment]
@Nimfadora , I restarted Travis failures, but please address other CIs failures.
I can not reproduce failure of `JavadocParseTreeTest#testInlineCustomJavadocTag()` on my local.

**Comment by Nimfadora** [comment]
@romani, @rnveach when I call mbn clean verify it fails with such exception: 
`[checkstyle] [ERROR] /home/valeria/ideaProjects/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java:38:1: Class 'AbstractFileSetCheckTest' must be declared as 'abstract'. [AbstractClassName]`
But it is a TEST class which tests class named AbstractFileSetCheck, there is no problem with AbstractCheckTest, but with my new test it throws exception, what should I do with it?

**Comment by romani** [comment]
> But it is a TEST class which tests class named AbstractFileSetCheck, there is no problem with AbstractCheckTest, but with my new test it throws exception, what should I do with it?

Look at checkstyle suprpessions - https://github.com/checkstyle/checkstyle/blob/master/config/suppressions.xml#L23

**Comment by romani** [comment]
@Nimfadora , please address points from rnveach.
please fix TC violations.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbsttractFileSetCheckTest.java (hunk 1 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbsttractFileSetCheckTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractCheckTest.java (hunk 2 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractCheckTest.java]: Fixed

**Comment by romani** [comment]
@Nimfadora , TC is still red, do you have problem with it ?

**Comment by Nimfadora** [comment]
@romani mvn clean verify passes localy, I don't understand what is wrong. Do you have any ideas? Why  does TC fail?

**Comment by romani** [comment]
> Do you have any ideas? Why does TC fail?

`mvn verify` and TC are completely independent validations. For TC issues please open IDEA , configure it and reproduce violations. Do you have no violations after launching IDEAs inspections ?

Travis failures are known problems , could be ignored, for now we do not know how to resolve such instability, non of us can reproduce it on local.

**Comment by romani** [comment]
Please also rebase on latest master, we did travis fix, it should pass.

**Pull request merged**